### PR TITLE
python39Packages.argh: fix tests

### PR DESCRIPTION
**Pull request by dotlambda**
###### Motivation for this change
https://github.com/neithere/argh/issues/148

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

cc @trofi

### CODE REVIEW

**Comment by mweinelt** [comment]
Unmaintained since 2016 *sigh*.

Dropped the unrequire argh dependency from barman. No further fallout visible.

**Comment by mweinelt** [comment]
We *could* fetch this patch: https://gitweb.gentoo.org/repo/gentoo.git/commit/dev-python/argh?id=6f194f12a2e30aad7da347848f7b0187e188f983

File: pkgs/development/python-modules/argh/default.nix [hunk 1]
```diff
@@ -29,6 +30,7 @@ buildPythonPackage rec {
     description = "An unobtrusive argparse wrapper with natural syntax";
     license = licenses.lgpl2;
     maintainers = with maintainers; [ domenkozar ];
+    broken = pythonAtLeast "3.9"; # https://github.com/neithere/argh/issues/148
```
Comment (jonringer) [inline pkgs/development/python-modules/argh/default.nix:33]: not use the `disabled = pythonAtLeast "3.9";`?

**Comment by dotlambda** [comment]
> We _could_ fetch this patch: https://gitweb.gentoo.org/repo/gentoo.git/commit/dev-python/argh?id=6f194f12a2e30aad7da347848f7b0187e188f983

Let's do that. Then we can also migrate `yrd` to Python 3.

**Review by trofi — APPROVED** [review]
The change fixes tests for me on `x86_64-linux`. Thank you!

**Pull request merged**